trieste/models/model_interfaces.py

```diff
@@ -537,6 +590,26 @@ def __init__(self, model: VGP, optimizer: Optimizer | None = None):
         super().__init__(optimizer)
         self._model = model
 
+        # GPflow stores num_data as a number. However, since we want to be able to update it
+        # without having to retrace the acquisition functions, put it in a Variable instead.
+        # So that the elbo method doesn't fail we also need to turn it into a property.
+        if not isinstance(self._model, VGPWrapper):
+            self._model._num_data = tf.Variable(
+                model.num_data or 0, trainable=False, dtype=tf.float64
+            )
+            self._model.__class__ = VGPWrapper
```

Comment:
Whats this doing?

Reply:
This turns num_data into a property backed by a tf.variable, since the gpflow elbo method expects to be able to access it as such. Just turning num_data into a variable fails for this reason.